### PR TITLE
fix some of the ufe headers promoted incorrectly.

### PR DESCRIPTION
**Pull request by HamedSabri-adsk**
Hi @ppt-adsk, I noticed accidentally some of the ufe header files are not promoted correctly according to the UFE version. 

Looking back in history I noticed this originated from one of my PRs. Je suis de soleil.

### CODE REVIEW

File: lib/mayaUsd/ufe/CMakeLists.txt [hunk 1]
```diff
@@ -70,18 +70,18 @@ set(headers
     UsdUndoRenameCommand.h
     Utils.h
     UfeVersionCompat.h
+    UsdAttribute.h
+    UsdAttributes.h
+    UsdContextOps.h
+    UsdObject3d.h
+    UsdUndoCreateGroupCommand.h
```
Comment (kxl-adsk) [inline lib/mayaUsd/ufe/CMakeLists.txt:77]: Only UsdUndoCreateGroupCommand.h seems to be V1. Everything else is V2 and should remain in the list below.

Comment (seando-adsk) [reply lib/mayaUsd/ufe/CMakeLists.txt:77]: Hamed, all of these are UFE V2 files and should remain where they are.

**Comment by HamedSabri-adsk** [comment]
Let me revisit this carefully.  I put this PR last night prematurely. sigh...

**Comment by HamedSabri-adsk** [comment]
> 
> 
> The real fix for the compilation of #388 with ufe v1 is to make conditional include of UsdUndoCreateGroupCommand in UsdHierarchy.cpp
> 
> ```
> #ifdef UFE_V2_FEATURES_AVAILABLE
> #include <mayaUsd/ufe/UsdUndoCreateGroupCommand.h>
> #endif
> ```

Indeed, Thanks @kxl-adsk .  I see now.......... What was I thinking? :) 

See b1c3c561ce0e7721ca06728f420c5482f9b8e32d

File: lib/mayaUsd/ufe/CMakeLists.txt (hunk 1 above)
Comment (HamedSabri-adsk) [reply lib/mayaUsd/ufe/CMakeLists.txt]: Thanks @seando-adsk . Indeed. I had a momentary brain damage when I made the change :)

**Pull request merged**